### ISSUE TITLE
Cannot read property 'protocol' of null

### ISSUE THREAD

**Issue by xppt** [fanout/websockhop#21]
Like in the previous issue (https://github.com/fanout/websockhop/issues/20) I wanted to check how WebSockHop will behave on connection lost and temporary added a blocking rule to my iptables list. After connection recover is saw an unhandled exception in this.socket.onopen handler:
```
websockhop.min.js:formatted:3711 Uncaught (in promise) TypeError: Cannot read property 'protocol' of null
    at e.<anonymous> (http://<some>/static/websockhop.min.js:278:4596)
    at e (http://<some>/static/websockhop.min.js:272:206)
    at Generator.136.n [as _invoke] (http://<some>/static/websockhop.min.js:272:1251)
    at Generator.136.t.forEach.t.(anonymous function) [as next] (http://<some>/static/websockhop.min.js:272:385)
    at n (http://<some>/static/websockhop.min.js:24:343)
    at 12.exports.default (http://<some>/static/websockhop.min.js:24:499)
    at Promise (<anonymous>)
    at r (http://<some>/static/websockhop.min.js:102:590)
    at WebSocket.12.exports.default (http://<some>/static/websockhop.min.js:24:277)
```

This is because of _raiseErrorEvent not properly closing socket (and not removing event handlers), just executing instead `this.socket = null`. In case of connection timeout some event can still happen, and will fail trying to access `this.socket`.

**Comment by gobengo** [comment]
@xppt I was able to reproduce this bug and made a fix at PR #23 . Can you confirm that that branch solves the problem in your project?

**Comment by jkarneges** [comment]
Merged.
